proposals/0024-repair-and-restart.md

```diff
@@ -0,0 +1,208 @@
+---
+simd: '0024'
+title: Repair and start in a cluster restart
+authors:
+  - Wen Xu (Solana Labs)
+category: Standard
+type: Core
+status: Draft
+created: 2023-04-07
+feature: (fill in with feature tracking issues once accepted)
+---
+
+## Summary
+
+Improve the current [cluster restart](https://docs.solana.com/running-validator/restart-cluster)
+procedure such that validators can automatically figure out the highest
+optimistically confirmed slot and then proceed to restart if everything looks
+fine.
+
+## New Terminology
+
+None
+
+## Motivation
+
+Currently during a cluster restart, validator operators need to decide latest
+optimistically confirmed slot, then restart the validators with new commandline
+arguments.
+
+The current process involves a lot of human intervention, if people make a
+mistake in deciding the highest optimistically confirmed slot, it could mean
+rollback of user transactions after those transactions have been confirmed,
+which is not acceptable.
+
+We aim to automate the finding of highest optimistically confirmed slot and
+block data distribution, so that we can lower the possibility of human mistakes
+in the cluster restart process.
+
+## Alternatives Considered
+
+See [Handling Common Solana Outages](https://docs.google.com/document/d/1RkNAyz-5aKvv5FF44b8SoKifChKB705y5SdcEoqMPIc)
+for details.
+
+There are many proposals about automatically detecting that the cluster is
+in an outage so validators should enter a recovery process automatically.
+
+While getting human out of the loop greatly improves recovery speed,
+automaticlly restarting the whole cluster still seems risky. Because if
+the recovery process itself doesn't work, it might be some time before
+we can get human's attention. And it doesn't solve the cases where new binary
+is needed. So for now we still plan to have human in the loop.
+
+## Detailed Design
+
+The new protocol tries to make all 80% restarting validators get the same
+data blocks and the same set of last votes among them, then they can probably
+make the same decision and then proceed.
+
+The steps roughly look like this:
+
+1. Everyone freezes, no new blocks, no new votes, and no Turbine
```

Comment:
"no turbine" seems oddly specific because it's just one protocol of many that run in the validator. 

Also this new recovery would be running completely separate of the validator, so turbine would be off by assumption.

Would be good to specify this is a protocol entirely separate from normal validator operation, i.e. step 1 is validator boots into a separate recovery mode.